### PR TITLE
Add Connections tab to list and manage (add/edit/remove) connections to a particular service

### LINKED ISSUE #38: Manage QGIS connections to PG using a particular service
The plugin could have another tab called, e.g., `Connections`.

In this tab we could have a combobox of services, and a list widget showing existing connections using the service.

I'm not sure if QGIS API allows it already, but we could allow users to add, edit or remove those connections, e.g., by using the QGIS `New connection` dialog.

This would help users to quickly and effortlessly create a QGIS connection to PG using the service they've just created or edited using the plugin.

### PR DESCRIPTION
**Pull request by gacarrillor**
(_External_ contribution here.) 

https://github.com/opengisch/qgis-pg-service-parser-plugin/assets/652785/3fd18bca-b16e-468c-affb-1fb27e531f9e

Add new Connections tab to:

  + List QGIS connections that use a particular service.
  + Allow users to effortlessly create a new QGIS connection using the service they've just created/edited.
  + Edit or remove QGIS connections to a particular service.

@m-kuhn, @3nids, @signedav, would this new tab make sense for you?

Fix #38

### CODE REVIEW

**Comment by signedav** [comment]
Looks nice and makes sense for sure. I wonder if it wouldn't have been easier to just link to the original data source manager instead (not sure if it's probided by the API though...)

**Comment by 3nids** [comment]
Sounds indeed useful, but I'm wondering if it might bring some confusion. The difference QGIS connection vs PG service is not super intuitive I think. Maybe it would be worth getting some input from @nirvn

**Comment by gacarrillor** [comment]
> Looks nice and makes sense for sure. I wonder if it wouldn't have been easier to just link to the original data source manager instead (not sure if it's probided by the API though...)

@signedav, thanks for your feedback. I'd really like to open the dialog to load layers, but couldn't find the way via API (in fact, the dialog opens but selected layers are never loaded, probably to missing SIGNAL/SLOT connections or something).

Nonetheless, note that this PR allows users to create their very first QGIS connection using a PG service with a couple of clicks, which is not always obvious, and actually, error prone (you need to type the service name), even having the `New Connection...` dialog open.

**Comment by m-kuhn** [comment]
This will indeed be a nice extension, to make up for the lack of a combobox or autocompletion for this field directly in QGIS.

In terms of usability, we could maybe think of unifying the "copy" and "edit" tabs under a single "services" tab and keep the "connections" tab where it is, that would make it clearer. I'd see this as a bigger change. For the shorter term, possibly renaming the first two tabs to "Edit service" and "Copy or new service" could be a way forward?

In any case, I'd not block this with too long UX discussions and rather merge this for the quick win :-)

**Comment by 3nids** [comment]
Maybe one proposal: merge the first 2 tabs into "PG service file" tab. For the copy, we just add a push button next to the combobox with `copy to new…` and `copy and overwirte …` (with the list of all services)

And we then have the "QGIS services" tab. What do you think?

Otherwise, I'm fine with merging like this.

**Comment by gacarrillor** [comment]
Summarizing naming proposals for tabs (and adding others):

1. `Edit`, `Copy`, `Connections`
2. `Services`, `Connections`
3. `Edit service`, `Copy or new service`, `Connections`
4. `PG service file`, `QGIS services`
5. `Edit`, `Copy`, `Connect`

I'd be happy to implement any of those, so let's reach an agreement through voting.
Please vote @m-kuhn, @3nids, @signedav, @ponceta and any other user that reads this :) (you can vote for multiple options or even add a new one).

Note that if an option to merge the first 2 tabs wins, we'll take that into account in a follow-up.

**Comment by 3nids** [comment]
Maybe "QGIS Connections" rather than services.
But let's merge this one first indeed.

File: pg_service_parser/ui/pg_service_dialog.ui [hunk 1]
```diff
@@ -326,6 +326,143 @@
        </item>
       </layout>
      </widget>
+     <widget class="QWidget" name="tabConnections">
+      <attribute name="title">
+       <string>Connections</string>
```
Comment (3nids) [inline pg_service_parser/ui/pg_service_dialog.ui:331]: QGIS Connections

**Comment by gacarrillor** [comment]
Rebased.

File: pg_service_parser/ui/pg_service_dialog.ui (hunk 1 above)
Comment (gacarrillor) [reply pg_service_parser/ui/pg_service_dialog.ui]: Done in https://github.com/opengisch/qgis-pg-service-parser-plugin/pull/39/commits/a0468a33be83f9f5c3f5053eb10e4ea9b32bc6cb

**Comment by gacarrillor** [comment]
> In terms of usability, we could maybe think of unifying the "copy" and "edit" tabs under a single "services" tab and keep the "connections" tab where it is, that would make it clearer. I'd see this as a bigger change. For the shorter term, possibly renaming the first two tabs to "Edit service" and "Copy or new service" could be a way forward?

Renamed the first two tabs to: `Edit Service` and `Copy Service`, to keep it short while we think about merging them.

**Pull request merged**